qCC/ccColorFromScalarDlg.cpp

```diff
@@ -332,29 +346,63 @@ void ccColorFromScalarDlg::updateHistogram(int n)
 void ccColorFromScalarDlg::updateChannel(int n)
 {
 	ccScalarField* sf = static_cast<ccScalarField*>(m_cloud->getScalarField(m_combos[n]->currentIndex()));
-	m_scalars[n] = sf;
 	if (sf)
 	{
-		sf->computeMinAndMax();
-		sf->setColorScale(m_colors[n]);
+		m_scalars[n] = sf;
+		m_scalars[n]->computeMinAndMax();
+		m_scalars[n]->setColorScale(m_colors[n]);
 
-		m_minSat[n] = sf->getMin();
-		m_maxSat[n] = sf->getMax();
+		m_minSat[n] = m_scalars[n]->getMin();
+		m_maxSat[n] = m_scalars[n]->getMax();
 		m_boxes_min[n]->setMinimum(m_minSat[n]);
 		m_boxes_min[n]->setMaximum(m_maxSat[n]);
 		m_boxes_max[n]->setMinimum(m_minSat[n]);
 		m_boxes_max[n]->setMaximum(m_maxSat[n]);
 		m_boxes_max[n]->setCorrectionMode(QAbstractSpinBox::CorrectionMode::CorrectToNearestValue);
+	}
+	if (m_scalars[n])
+	{
 		updateHistogram(n);
-
-		//set default stretch (n.b. this is a cheap hack to avoid calculating percentiles [by assuming uniform data distribution])
-		float range = m_maxSat[n] - m_minSat[n];
-		m_histograms[n]->setMinSatValue(m_minSat[n] + 0.1 * range);
-		m_histograms[n]->setMaxSatValue(m_maxSat[n] - 0.1 * range);
 	}
 }
 
-//mapping ranges changed
+void ccColorFromScalarDlg::setDefaultSatValuePerChannel(int n)
+{
+	//set default stretch (n.b. this is a cheap hack to avoid calculating percentiles [by assuming uniform data distribution])
+	m_scalars[n]->setColorScale(m_colors[n]);
+	m_minSat[n] = m_scalars[n]->getMin();
+	m_maxSat[n] = m_scalars[n]->getMax();
+	float range = m_maxSat[n] - m_minSat[n];
+	m_histograms[n]->setMinSatValue(m_minSat[n] + 0.1 * range);
+	m_histograms[n]->setMaxSatValue(m_maxSat[n] - 0.1 * range);
+}
+
+void ccColorFromScalarDlg::jiggleSatValuePerChannel(int n)
+{
+	//Ugly hack to force the plots saturation values to refresh
+	//properly after resize
+	m_scalars[n]->setColorScale(m_colors[n]);
+	double minSat = m_boxes_min[n]->value();
+	double maxSat = m_boxes_max[n]->value();
+	m_histograms[n]->setMinSatValue(minSat + 0.1);
+	m_histograms[n]->setMaxSatValue(maxSat - 0.1);
+	m_histograms[n]->setMinSatValue(minSat - 0.1);
+	m_histograms[n]->setMaxSatValue(maxSat + 0.1);
+	m_histograms[n]->replot();
```

Comment:
Is it necessary to call both replot and repaint? And can't we change the ccHistogram class so as to avoid needing to 'jiggle' these values ;) ?

Reply:
Honestly no idea, I called it an ugly hack because I tried a lot of different things that seemed like they should work, but didn't... and eventually settled on this. I think I spent most of yesterday messing around with this PR. 
I would prefer anything other than this 'jiggle' values hack, but I'm not sure what causes the issue in the first place considering during resize I would have expected QCustomPlot to take care of it. maybe the new version Andy setup will help with that?

Reply:
It's possible. There are a lot of changes - the old one is almost 5 years old. And there were changes to the way drawing was done.

I made those changes on master because I thought they were too large for 2.11. I would suggest we (1) sort out the conflicting PRs/solutions, then (2) you try to apply your changes on master w/the updated custom plot.

If that works better than we can look at bringing those changes back to 2.11.